query/src/sql/planner/binder/bind_context.rs

```diff
@@ -43,13 +43,19 @@ pub struct ColumnBinding {
 #[derive(Clone, Default)]
 pub struct BindContext {
     _parent: Option<Box<BindContext>>,
-    columns: Vec<ColumnBinding>,
+    pub columns: Vec<ColumnBinding>,
 
     /// The relational operator in current context
     pub expression: Option<SExpr>,
 
     /// Aggregation scalar expression
     pub agg_scalar_exprs: Option<Vec<Scalar>>,
+
+    /// Order by columnBinding, consider the sql: select sum(a) from t group by a,b order by b;,
+    /// Order by requires not just the columns in the selection,
+    /// but the columns of the entire table as well as the columns of the selection
+    ///
```

Comment:
I updated the annotation @leiysky 

The `bind_select_stmt` method returns a `BindContext` whose columns only contain colums of selection, but for the following case:
```sql
mysql> create table t(a int, b int);
Query OK, 0 rows affected (0.03 sec)
Read 0 rows, 0.00 B in 0.005 sec., 0 rows/sec., 0.00 B/sec.

mysql> insert into t values(1, 2),(2,3)
    -> ;
Query OK, 0 rows affected (0.04 sec)
Read 2 rows, 16.00 B in 0.011 sec., 184.19 rows/sec., 1.44 KiB/sec.
mysql> select sum(a) from t group by a,b order by b;
+--------+
| sum(a) |
+--------+
|      1 |
|      2 |
+--------+
2 rows in set (0.02 sec)
Read 2 rows, 16.00 B in 0.006 sec., 310.92 rows/sec., 2.43 KiB/sec.
```
`Order by` alao need the column of `a`, `b`, so in the case `order_by_columns` is used to store `a`, `b`, `sum(a)`

Reply:
I think it's better to pass a `has_order_by` hint to `bind_select_stmt`, and associate it with related code.